### PR TITLE
Fix team edits delete students

### PR DESCRIPTION
**Pull request by gibsonbailey**
**Don't merge this until we update the version of DRF writable**, since that's the solution to students being deleted.

- [x] Marking this ready to merge once drf-writable-nested is updated in the requirements. Will point this to my fork until it's merged and tagged over at the drf-writable-nested repo. @Tthomas63

### CODE REVIEW

File: apps/api/tests/test_teams.py [hunk 1]
```diff
@@ -113,3 +113,19 @@ def test_instructor_user_can_only_get_and_post_teams(self):
...
+            reverse('api:team-detail', kwargs={'version': 'v1', 'pk': self.team.pk}),
+            data=json.dumps({
+                "klass": self.klass.pk,
+                "members": []
+            }),
+            content_type='application/json'
+        )
+        assert resp.data['name'] == 'test'
+        assert resp.status_code == 200
+        assert initial_student_count == StudentMembership.objects.count()
```
Comment (gibsonbailey) [inline apps/api/tests/test_teams.py:131]: Line 131 is the important one.

File: static/riotjs/wizard/create_team.tag [hunk 2]
```diff
@@ -14,17 +14,19 @@
         <div class="fields">
             <div class="eight wide field">
                 <label>Members:</label>
```
Comment (gibsonbailey) [inline static/riotjs/wizard/create_team.tag:16]: These fields now use Semantic UI: a multiple select dropdown for members and a single select dropdown for leaders.

File: static/riotjs/wizard/create_team.tag [hunk 3]
```diff
@@ -104,13 +94,35 @@
...
                 })
                 .fail(function (response) {
-                    console.log(response)
                     Object.keys(response.responseJSON).forEach(function (key) {
                         toastr.error("Error with " + key + "! " + response.responseJSON[key])
                     });
                 })
         }
 
+        self.get_eligible_leaders = () => {
```
Comment (gibsonbailey) [inline static/riotjs/wizard/create_team.tag:103]: Only list team members in the leader dropdown.

File: static/riotjs/wizard/create_team.tag [hunk 4]
```diff
@@ -104,13 +94,35 @@
...
+                    }
+                })
+        }
+
+        self.leader_dropdown_on_change = (value, text, $selectedItem) => {
+            let leader_index = self.klass.enrolled_students.findIndex( (student) => {
+                return student.id == value
+            })
+
+            for (let i = 0; i < self.klass.enrolled_students.length; i++) {
```
Comment (gibsonbailey) [inline static/riotjs/wizard/create_team.tag:121]: Mark the new student as the leader on the `enrolled_students` array.

File: apps/api/tests/test_teams.py (hunk 1 above)
Comment (Tthomas63) [reply apps/api/tests/test_teams.py:131]: 👍

File: static/riotjs/wizard/create_team.tag (hunk 2 above)
Comment (Tthomas63) [reply static/riotjs/wizard/create_team.tag:16]: 👍

File: static/riotjs/wizard/create_team.tag (hunk 3 above)
Comment (Tthomas63) [reply static/riotjs/wizard/create_team.tag:103]: 👍

File: static/riotjs/wizard/create_team.tag (hunk 4 above)
Comment (Tthomas63) [reply static/riotjs/wizard/create_team.tag:121]: 👍

**Review by Tthomas63 — APPROVED** [review]
:shipit:

**Pull request merged**